src/model/schema.js

```diff
@@ -581,3 +553,45 @@ export class SchemaItem {
  * @typedef {String|Array.<String|module:engine/model/element~Element>|module:engine/model/position~Position}
  * module:engine/model/schema~SchemaPath
  */
+
+// Checks whether the given checkPath and allowedPath right ends match.
+//
+// checkPath: C, D
+// allowedPath: A, B, C, D
+// result: OK
+//
+// checkPath: A, B, C
+// allowedPath: A, B, C, D
+// result: NOK
+//
+// Note – when matching paths, element extension chains (inheritance) are taken into consideration.
+//
+// @param {Schema} schema
+// @param {Array.<String>} checkPath
+// @param {Array.<String>} allowedPath
+function matchPaths( schema, checkPath, allowedPath ) {
+	// Start checking from the right end of both tables.
+	let allowedPathIndex = allowedPath.length - 1;
+	let checkPathIndex = checkPath.length - 1;
+
+	// And finish once reaching an end of the shorter table.
```

Comment:
Okay it took me a while to analyze this. At first sight it seemed wrong to me that any path can be shorter. For `checkPath` = `C D E` both `allowedPath` = `A B C D E` and `allowedPath` = `D E` are right. This rises a question whether `allowedPath` should be matched in whole? In first `allowedPath` it seems that somebody explicitly want to set that `E` is allowed in `A B C D`, that is that one of ancestors is `A`.

OTOH, this will not be a problem if in `checkPath` we will always pass a "full path", that is a path that has `$root` as left-most element. This would "solve" the "issue" described above. Consider again:
`checkPath` = `$root C D E`, `allowedPath` = `A B C D E` and `allowedPath` = `$root A B C D E`.

Now, check will fail for both `allowedPath`s (which allows us to use shorter and simpler form without `$root`).

And now back to the original problem. If we suppose that `checkPath` always has `$root` as first element, a check should fail for every `checkPath` shorter than `allowedPath`. That is because there will be a moment when we will check `$root` from `checkPath` against non-`$root` from `allowedPath`.

So there are a few solutions:
1. We leave code that is here as it is and stress in documentation that `checkPath` has to be full. Drawback: it's prone to generate bugs. OTOH, I expect that in most cases, inside will be specified as `model.Position`.
2. Automatically add `$root` as first element to `checkPath` if first element is not `$root`. Drawback: this may backfire if, for a reason, really want to check a path that starts from something else than `$root`.
3. Return `false` if `checkPath` is shorter than `allowedPath`. This feels natural, because it means that it is impossible to match every item from explicitly stated `allowedPath` that should require, that the last item in path is exactly in those parents. Drawbacks: IDK?